Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/Controller.purs

```diff
@@ -842,14 +862,33 @@ eval (MessagingViewActionController (MessagingView.SendSuggestion chatSuggestion
     continue state {data {chatSuggestionsList = []}, props {canSendSuggestion = false}}
   else continue state
 
+eval (MessagingViewActionController (MessagingView.ToggleMultiChatPopUp)) state = do 
+  if state.props.showChatListPopUp then do 
+    continue state {props {showChatListPopUp = false}}
+  else 
+    do
+      void $ pure $ hideKeyboardOnNavigation true
+      continue state {props {showChatListPopUp = true}}
+
+eval (MessagingViewActionController (MessagingView.SwitchChat contact)) state = do
+  let chatPersonId = if contact.recipient == CMC.DRIVER && state.props.currentStage /= RideStarted then "Customer" else (getValueFromCache (show CUSTOMER_ID) getKeyInSharedPrefKeys)
```

Comment:
can we store the correct values in contact? so we don't need to add checks here.